src/badge.jsx

```diff
@@ -0,0 +1,110 @@
+const React = require('react');
+const Typography = require('./styles/typography');
+const DefaultRawTheme = require('./styles/raw-themes/light-raw-theme');
+const ThemeManager = require('./styles/theme-manager');
+
+// Badge
+export default React.createClass({
+  displayName: 'Badge',
+  contextTypes: {
+    muiTheme: React.PropTypes.object,
+  },
+  //for passing default theme context to children
+  childContextTypes: {
+    muiTheme: React.PropTypes.object,
+  },
+  getChildContext () {
+    return {
+      muiTheme: this.state.muiTheme,
+    };
+  },
+  propTypes: {
+    className: React.PropTypes.string,
+    primary: React.PropTypes.bool,
+    secondary: React.PropTypes.bool,
+    labelColor: React.PropTypes.string,
+    backgroundColor: React.PropTypes.string,
+    style: React.PropTypes.object,
+    primaryText: React.PropTypes.node.isRequired,
+  },
+  getInitialState() {
+    return {
+      muiTheme: this.context.muiTheme ? this.context.muiTheme : ThemeManager.getMuiTheme(DefaultRawTheme),
+    };
+  },
+  getDefaultProps() {
+    return {
+      className: '',
+      primary: false,
+      secondary: false,
+      labelColor: null,
+      backgroundColor: null,
+      style: null,
+    };
+  },
+  componentWillReceiveProps(nextProps, nextContext) {
+    let newMuiTheme = nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme;
+    this.setState({
+      muiTheme: newMuiTheme,
+    });
+  },
+  getStyles() {
+    const theme = this.state.muiTheme.badge;
+
+    const backgroundColor = this.props.backgroundColor
+      ? this.props.backgroundColor
+      : this.props.primary
+        ? theme.primaryColor
+        : this.props.secondary
+          ? theme.secondaryColor
+          : theme.color;
+
+    const labelColor = this.props.labelColor
+      ? this.props.labelColor
+        : this.props.primary
+          ? theme.primaryTextColor
+          : this.props.secondary
+            ? theme.secondaryTextColor
+            : theme.textColor;
+
+    const style = {
+      root: {
+        backgroundColor: 'none',
```

Comment:
This is not a valid CSS value. See http://stackoverflow.com/questions/8739665/is-background-colornone-valid-css. What's do you try to acheive?